### DISCUSSION TITLE
Error: ENOENT: no such file or directory .lowdefy/dev/.next/BUILD_ID

### DISCUSSION THREAD

**Discussion by sivadotblog** [lowdefy/lowdefy#1819] [Questions and Help]
This is my first time trying my hands on lowdefy. I am this on running windows and wsl2.  I cant seem to get past this build error. since its a pnpx package am not sure why its asking me for the build id

```
npx lowdefy@4 dev
10:08:14 ∙ Running 'lowdefy dev'. Lowdefy app version 4.1.0.
10:08:14 ℹ Starting development server.
10:14:02 ∙ Dependencies install successfully.
10:14:03 ∙ > @lowdefy/server-dev@4.1.0 start /mnt/c/Users/sivan/code/lowdefy_project/.lowdefy/dev
10:14:03 ∙ > node manager/run.mjs
10:14:31 ∙ fatal: not a git repository (or any parent up to mount point /mnt)
10:14:31 ∙ Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
10:14:31 ⚠ No menus found. Building default menu.
10:14:32 ∙ Built config.
10:14:38 ∙ Installed plugins.
10:21:27 ✖ node exited with code 1
10:21:42 ✖ [Error: ENOENT: no such file or directory, open '/mnt/c/Users/sivan/code/lowdefy_project/.lowdefy/dev/.next/BUILD_ID'] {
10:21:42 ✖   errno: -2,
10:21:42 ✖   code: 'ENOENT',
10:21:42 ✖   syscall: 'open',
10:21:42 ✖   path: '/mnt/c/Users/sivan/code/lowdefy_project/.lowdefy/dev/.next/BUILD_ID'
10:21:42 ✖ }

```
should i run build?

**Comment by cdgipson** [comment]
I've run into the same issue, currently I have no resolution.

**Reply by SamTolmay** [reply]
Hi @cdgipson 

Could you please run
```
pnpx lowdefy@dev --log-level=debug
```
and let me know what output you get?

**Comment by cdgipson** [comment]
PS C:\repos\lowdefy> pnpx ***@***.*** --log-level=debug
C:\Users\cgipson\AppData\Local\pnpm-cache\dlx\gb4mwbm2w3g3blma6bfswti57m\1926785ba22-acc0:
 ERR_PNPM_NO_MATCHING_VERSION  No matching version found for ***@***.***

This error happened while installing a direct dependency of C:\Users\cgipson\AppData\Local\pnpm-cache\dlx\gb4mwbm2w3g3blma6bfswti57m\1926785ba22-acc0

The latest release of lowdefy is "4.3.0".

Other releases are:
  * alpha: 4.0.0-alpha.37
  * rc: 4.0.0-rc.12
  * experimental: 0.0.0-experimental-20240115131502

**Reply by cdgipson** [reply]
I was able to get lowdefy running when working via WSL Ubuntu.  Not ideal, but it works.

**Reply by Gervwyk** [reply]
Hi @sivadotblog, just a correction on @SamTolmay comment. If you run `pnpx lowdefy@4 dev --log-level=debug` it should provide more detail as to the issue. Glad it's working. Would be interested to figure out the root cause.

**Reply by cdgipson** [reply]
```
PS C:\repos\lowdefy> pnpx lowdefy@4 dev --log-level=debug
09:05:33 ∙ Running 'lowdefy dev'. Lowdefy app version 4.3.0.
09:05:33 ℹ Starting development server.
09:05:36 ∙ Installing dependencies.
09:05:36 + Lockfile is up to date, resolution step is skipped
09:05:36 + Already up to date
09:05:42 + Done in 7.5s
09:05:42 ∙ Dependencies install successfully.
09:05:43 ∙ > @lowdefy/server-dev@4.3.0 start C:\repos\lowdefy\.lowdefy\dev
09:05:43 ∙ > node manager/run.mjs
09:05:56 ∙ fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree.
09:05:56 ∙ Use '--' to separate paths from revisions, like this:
09:05:56 ∙ 'git <command> [<revision>...] -- [<file>...]'
09:05:56 ⚠ No menus found. Building default menu.
09:05:57 ∙ Built config.
09:05:57 ✖ spawn EINVAL
09:05:57 + Started next server with pid 9072.
09:06:05 ✖ Error: ENOENT: no such file or directory, open 'C:\repos\lowdefy\.lowdefy\dev\.next\BUILD_ID'
09:06:05 ✖     at async open (node:internal/fs/promises:641:25)
09:06:05 ✖     at async Object.readFile (node:internal/fs/promises:1254:14)
09:06:05 ✖     at async setupFsCheck (C:\repos\lowdefy\.lowdefy\dev\node_modules\.pnpm\next@13.5.4_react-dom@18.2.0_react@18.2.0__react@18.2.0\node_modules\next\dist\server\lib\router-utils\filesystem.js:142:19)
09:06:05 ✖     at async initialize (C:\repos\lowdefy\.lowdefy\dev\node_modules\.pnpm\next@13.5.4_react-dom@18.2.0_react@18.2.0__react@18.2.0\node_modules\next\dist\server\lib\router-server.js:55:23)
09:06:05 ✖     at async Server.<anonymous> (C:\repos\lowdefy\.lowdefy\dev\node_modules\.pnpm\next@13.5.4_react-dom@18.2.0_react@18.2.0__react@18.2.0\node_modules\next\dist\server\lib\start-server.js:219:36) {
09:06:05 ✖   errno: -4058,
09:06:05 ✖   code: 'ENOENT',
09:06:05 ✖   syscall: 'open',
09:06:05 ✖   path: 'C:\\repos\\lowdefy\\.lowdefy\\dev\\.next\\BUILD_ID'
09:06:05 ✖ }
09:06:05 + nextServer exit 9072, signal: null, code: 1
```

**Reply by machielvdw** [reply]
Hi @cdgipson. Sometimes issues like this can be because of a Node.js version. You can run `node -v` to check this, I recommend using LTS (currently v20.18.0).

**Reply by cdgipson** [reply]
Sorry for the late response.  I was running the latest version of Node, under Windows 11.

**Reply by diegobrum** [reply]
I have same problem
Doesn't work in node versions 18, 20 and 22, but works fine in version 21.6.1

**Reply by cdgipson** [reply]
I will give that version of Node a try.

**Comment by machielvdw** [comment]
Hi @sivadotblog, @cdgipson, @diegobrum.
We have released Lowdefy version 4.4.0 that fixes this issue. You can update your app to this version. Thanks.

https://github.com/lowdefy/lowdefy/releases/tag/v4.4.0

**Reply by cdgipson** [reply]
I updated the application, now it is stuck with the following error...which makes no sense.



    PS C:\repos\lowdefy> pnpm add -g lowdefy
    Packages: +94
    ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
    Progress: resolved 94, reused 94, downloaded 0, added 94, done

    C:\Users\cgipson\AppData\Local\pnpm\global\5:
    + lowdefy 4.4.0

    Done in 11.6s
    PS C:\repos\lowdefy> lowdefy init
    09:11:52 ✖
    -------------------------------------------------------------
      The package manager "pnpm" is required to run Lowdefy.
      Install pnpm as describe here:
      https://pnpm.io/installation
      or run  'corepack enable'.
    -------------------------------------------------------------
    09:11:52 ✖ The package manager "pnpm" is required to run Lowdefy.


You can see that pnpm has been installed, because I was able to install lowdefy cli.
